sdk/search/Azure.Search/src/SearchIndexClient.cs

```diff
@@ -198,7 +205,8 @@ public class SearchIndexClient
             Debug.Assert(!string.IsNullOrEmpty(indexName));
             Debug.Assert(pipeline != null);
             Debug.Assert(diagnostics != null);
-            Debug.Assert(SearchClientOptions.ServiceVersion.V2019_05_06 <= version &&
```

Comment:
Shouldn't this have been "asserted" (with exceptions) higher up the callstack already?

Reply:
I (try to) use assertions at public facing boundaries and Debug at private boundaries.  This is just a sanity check we should never hit.